Python/gc_free_threading.c

```diff
@@ -350,8 +391,8 @@ process_delayed_frees(PyInterpreterState *interp)
 }
 
 // Subtract an incoming reference from the computed "gc_refs" refcount.
-static int
-visit_decref(PyObject *op, void *arg)
```

Comment:
The GC and frame visiting need to be coupled because the standard `tp_visit` pattern doesn't work correctly with deferred references. You need to account for deferred references differently in `update_refs` than in `mark_reachable`:

* Deferred references should not "subtract one" from the computed reference count (`update_refs`)
* However, they should count for the transitive reachability check  (`mark_reachable`)

We can implement it slightly differently by pushing frame visiting to the GC implementation rather than exposing `_Py_visit_decref` outside of the GC implementation.